libcloudforensics/providers/gcp/forensics.py

```diff
@@ -430,3 +432,63 @@ def VMRemoveServiceAccount(project_id: str,
     logger.error('Fatal exception encountered: {0:s}'.format(str(exception)))
 
   return True
+
+
+def CheckInstanceSSHAuth(project_id: str,
+                         instance_name: str) -> Optional[List[str]]:
+  """Check enabled SSH authentication methods for an instance.
+
+  Uses SSH with the verbose flag to check enabled SSH authentication methods.
+  Note this Will generate a log line on the target host:
+
+    "Connection closed by authenticating user root <your_ip> port <port>
+    [preauth]"
+
+  Args:
+    project_id (str): the project id for the instance.
+    instance_name (str): the instance name to check.
+
+  Returns:
+    List[str]: The SSH authentication methods supported by the instance or
+      None if SSH wasn't accessible.
+  """
+  ssh_auth_pattern = re.compile(r'Authentications that can continue: (.*)')
+  # Arguments intended to prevent SSH from picking up any configuration from
+  # the host environment.
+  ssh_args = ['/usr/bin/ssh',
+              '-v',
+              '-F', '/dev/null',
+              '-oIdentitiesOnly=yes',
+              '-oIdentityFile=/dev/null',
+              '-oIdentityAgent=none',
+              '-oPKCS11Provider=none',
+              '-oSecurityKeyProvider=none',
+              '-oStrictHostKeyChecking=no',
+              '-oNumberOfPasswordPrompts=0',
+              '-oUserKnownHostsFile=/dev/null']
+
+  project = gcp_project.GoogleCloudProject(project_id)
+  instance = project.compute.GetInstance(instance_name)
+  instance_info = instance.GetOperation()
+  networks = instance_info.get('networkInterfaces', [])
+
+  external_ips = []
+  for network in networks:
+    if 'accessConfigs' in network:
+      for config in network['accessConfigs']:
```

Comment:
```suggestion
    for config in network.get('accessConfigs'):
```
maybe you can use get inn the loop and get rid of the if condition before

Reply:
Nice! That is more concise, change made under the new function in GoogleCloudCompute.